pkg/runtime/hotreload/hotreload.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hotreload
+
+import (
+	"context"
+	"errors"
+
+	compapi "github.com/dapr/dapr/pkg/apis/components/v1alpha1"
+	httpendapi "github.com/dapr/dapr/pkg/apis/httpEndpoint/v1alpha1"
+	"github.com/dapr/dapr/pkg/concurrency"
+	"github.com/dapr/dapr/pkg/config"
+	operatorpb "github.com/dapr/dapr/pkg/proto/operator/v1"
+	"github.com/dapr/dapr/pkg/runtime/authorizer"
+	"github.com/dapr/dapr/pkg/runtime/channels"
+	"github.com/dapr/dapr/pkg/runtime/compstore"
+	"github.com/dapr/dapr/pkg/runtime/hotreload/loader/disk"
+	"github.com/dapr/dapr/pkg/runtime/hotreload/loader/operator"
+	"github.com/dapr/dapr/pkg/runtime/hotreload/reconciler"
+	"github.com/dapr/dapr/pkg/runtime/processor"
+	"github.com/dapr/kit/logger"
+)
+
+var log = logger.NewLogger("dapr.runtime.hotreload")
+
+type OptionsDisk struct {
+	Config         *config.Configuration
+	Dirs           []string
```

Comment:
Is it config dir? What does it mean?

Reply:
The list of directories on disk which contain the yaml files containing the component and HTTP endpoints which need to be watched in order to respond to updates and hot-reload.

Reply:
This might be worth adding as a comment? Also, the distinction on the 2 structs